pkg/stanza/fileconsumer/config.go

```diff
@@ -182,6 +185,12 @@ func (c Config) validate() error {
 		return fmt.Errorf("`delete_after_read` cannot be used with `start_at: end`")
 	}
 
+	// Poll file limit can be 0 to signal unlimited or it must be greater than or equal to
+	// max concurrent files to ensure we can read at least the configured max concurrent files in a polling cycle.
+	if c.PollFileLimit < 0 || (c.PollFileLimit < c.MaxConcurrentFiles && c.PollFileLimit != 0) {
+		return fmt.Errorf("`poll_file_limit` must be 0 or greater than or equal to `max_concurrent_files`")
+	}
```

Comment:
Can we separate this into two checks/errors? The user will only violate one or the other at a time, so having a more specific message for each case would be clearer.